src/transformers/models/llava/processing_llava.py

```diff
@@ -104,10 +111,38 @@ def __call__(
             pixel_values = self.image_processor(images, return_tensors=return_tensors)["pixel_values"]
         else:
             pixel_values = None
+
+        if isinstance(text, str):
+            text = [text]
+        elif not isinstance(text, list) and not isinstance(text[0], str):
+            raise ValueError("Invalid input text. Please provide a string, or a list of strings")
+
+        # try to expand inputs in processing if we have the necessary parts
+        if self.patch_size is not None and self.vision_feature_select_strategy is not None:
+            # Replace the image token with the expanded image token sequence
+            num_image_tokens = (self.image_size // self.patch_size) ** 2 + 1
+            if self.vision_feature_select_strategy == "default":
+                num_image_tokens -= 1
+
+            prompt_strings = []
+            for sample in text:
+                sample = sample.replace(self.image_token, self.image_token * num_image_tokens)
+                prompt_strings.append(sample)
+        else:
+            prompt_strings = text
+            logger.warning_once(
+                "Expanding inputs for image tokens in LLaVa should be done in processing. "
+                "Please add `patch_size` and `vision_feature_select_strategy` to the model's image processing config. "
+                "Using processors without these attributes in the config is deprecated and will throw an error in v4.44."
+            )
```

Comment:
I think this is both going to be annoying to handle and creates leakiness in our abstractions: not everyone who uses the image processor will be using it for llava, and there's no reason for image processors for models which don't have `patch_size` to have this attribute. 

Instead, I'd suggest getting the defaults from the image processor if available, but also allowing them as processor arguments in the init and possibly the forward pass which the user can directly set and control. This way, when we save out the processor, these can be saved alongside, and the user doesn't need to change an image processor or override the processor each time they want to use it

Reply:
Right, that will be easier for users. Will make changes